sw/device/lib/base/BUILD

```diff
@@ -110,6 +110,27 @@ opentitan_functest(
     ],
 )
 
+cc_library(
+    name = "memory_prefixed",
+    srcs = ["memory.c"],
+    hdrs = ["memory.h"],
+    defines = [
+        "OT_PREFIX_MEM_FUNCTIONS",
```

Comment:
I'd like to avoid this semi-duplicate of the `:memory` target. Is there some Bazel magic that will enable us to `#define` in the `:memory_unittest` target and have it cascade down to the `:memory` dep?

Reply:
That's not really possible in bazel (technically it is possible via a transition to cause a re-evaluation of the build graph under the transitioning node, but that's way to complex for what you're trying to accomplish here.  The real use-case for a transition is a major configuration change, like setting the target architecture).

We have occasionally encountered this problem - the solution you have is about correct.  If the only user of `:memory_prefixed` is the unit tests, then you may want to just migrate the srcs, deps and defines into the unit test's srcs, deps and defines.